WEB-INF/classes/gov/noaa/pfel/erddap/handlers/HandlerFactory.java

```diff
@@ -32,99 +35,51 @@ public static State getHandlerFor(
-          "EDDTableFromWFSFiles" -> {
-        return new EDDTableFromFilesHandler(saxHandler, datasetID, completeState, datasetType);
-      }
-      case "EDDGridAggregateExistingDimension" -> {
-        return new EDDGridAggregateExistingDimensionHandler(
-            saxHandler, datasetID, completeState, context);
-      }
-      case "EDDGridCopy" -> {
-        return new EDDGridCopyHandler(saxHandler, datasetID, completeState, context);
-      }
-      case "EDDGridFromEDDTable" -> {
-        return new EDDGridFromEDDTableHandler(saxHandler, datasetID, completeState, context);
-      }
-      case "EDDGridLon0360" -> {
-        return new EDDGridLon0360Handler(saxHandler, datasetID, completeState, context);
-      }
-      case "EDDGridSideBySide" -> {
-        return new EDDGridSideBySideHandler(saxHandler, datasetID, completeState, context);
-      }
-      case "EDDTableAggregateRows" -> {
-        return new EDDTableAggregateRowsHandler(saxHandler, datasetID, completeState, context);
-      }
-      case "EDDTableCopy" -> {
-        return new EDDTableCopyHandler(saxHandler, datasetID, completeState, context);
-      }
-      case "EDDTableFromCassandra" -> {
-        return new EDDTableFromCassandraHandler(saxHandler, datasetID, completeState);
-      }
-      case "EDDTableFromDapSequence" -> {
-        return new EDDTableFromDapSequenceHandler(saxHandler, datasetID, completeState);
-      }
-      case "EDDTableFromDatabase" -> {
-        return new EDDTableFromDatabaseHandler(saxHandler, datasetID, completeState);
-      }
-      case "EDDTableFromAsciiService" -> {
-        return new EDDTableFromAsciiServiceHandler(
-            saxHandler, datasetID, completeState, datasetType);
-      }
-      case "EDDTableFromOBIS" -> {
-        return new EDDTableFromOBISHandler(saxHandler, datasetID, completeState);
-      }
-      case "EDDTableFromSOS" -> {
-        return new EDDTableFromSOSHandler(saxHandler, datasetID, completeState);
-      }
-      case "EDDTableFromFileNames" -> {
-        return new EDDTableFromFileNamesHandler(saxHandler, datasetID, completeState);
-      }
-      case "EDDGridFromAudioFiles",
-          "EDDGridFromNcFiles",
-          "EDDGridFromNcFilesUnpacked",
-          "EDDGridFromMergeIRFiles" -> {
-        return new EDDGridFromFilesHandler(saxHandler, datasetID, completeState, datasetType);
-      }
-      case "EDDGridFromEtopo" -> {
-        return new EDDGridFromEtopoHandler(saxHandler, datasetID, completeState);
-      }
-      default -> {
-        nTry--;
-        context.getNTryAndDatasets()[0] = nTry;
-        throw new IllegalArgumentException("Unknown dataset type: " + datasetType);
+
+    // TODO using the first constructor here, should we scan through all to find the most
+    // appropriate one?
+    Constructor<?> constructor = eddClassInfo.getSaxHandlerClass().get().getConstructors()[0];
+
+    // Constructors for handlers don't have uniform signatures so we have to do some investigation
```

Comment:
Invoking the constructor on the sax handler/State class is complex because the various State subclasses/Handlers don't have a uniform constructor signature.

Reply:
If I remember correctly there's essentially 2 different constructors for the dataset handlers. Many take 3 parameters (saxHandler, datasetId, completeState) and some take a 4th parameter (SaxParsingContext). It could make sense to have a dataset constructor param object that has all 4 of those and have all datasets take that object in their constructor.

Reply:
Looks like there are currently three constructor signature variations with five total possible arguments, and the first three are always consistent.

```
git show main:WEB-INF/classes/gov/noaa/pfel/erddap/handlers/HandlerFactory.java | grep -o "saxHandler, datasetID,[^)]*" | sort | uniq -c
     10 saxHandler, datasetID, completeState
      9 saxHandler, datasetID, completeState, context
      3 saxHandler, datasetID, completeState, datasetType
```

Reply:
Ah, I forgot the FromFilesHandler version. dataset type can be included in that param object.